utils_test.go

```diff
@@ -340,10 +340,75 @@ func HTTPResponseEndToEnd() *httptest.Server {
 
 }
 
+// HTTPResponseStubAndMirrorRequest Stubbing HTTP response
+func HTTPResponseStubAndMirrorCodeRequest() *httptest.Server {
+	return httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+
+		body, _ := ioutil.ReadAll(r.Body)
+
+		if r.Method == "POST" && len(body) == 0 {
+			log.Fatal("Missing Body on POST request")
+		} else if r.Method == "GET" && len(body) > 0 {
+			log.Fatal("Unexpected body in GET request")
+		}
+
+		resp := &mirroredResponse{
+			URL:    r.RequestURI,
+			Method: r.Method,
+			Token:  r.Header.Get("Authorization"),
+			Body:   body,
+		}
+		marshalledResp, _ := json.Marshal(resp)
+		w.Write(marshalledResp)
+	}))
+}
+
 func readFixture(path string) []byte {
 	data, err := ioutil.ReadFile(path)
 	if err != nil {
 		panic(err)
 	}
 	return data
 }
+
+func HTTPResponseCodeIssueStubAndMirrorRequest() *httptest.Server {
+
+	var resp []byte
+	return httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+
+		if r.RequestURI == "" {
+			resp = []byte("404 - url mismatch")
+		} else if r.RequestURI == "/v1/org/123/project/456/issue/xxbac5ed-83dd-xx65-8730-2xxx4467e00d/jira-issue" {
+			resp = readFixture("./fixtures/snyk_code_fixtures/results/singleCodeJiraTicketOpeningResponse")
+		} else if r.RequestURI == "/v3/orgs/123/issues?project_id=456&version=2021-08-20~experimental" {
```

Comment:
if we use, 2021-08-20~experimental how do we know that this tool will keep working with the latest revision of the API? hard to know without testing the live version I guess....